File: solutions/advent_of_code/2022/Day 1/part2.cpp
// Advent of Code Problem Link: https://adventofcode.com/2022/day/1

#include <bits/stdc++.h>
using namespace std;

array<int, 3> MAX = {-1, -1, -1};

void mx(int num) {
  if (num > MAX[0]) {
    MAX[2] = MAX[1];
    MAX[1] = MAX[0];
    MAX[0] = num;
  } else if (num > MAX[1]) {
    MAX[2] = MAX[1];
    MAX[1] = num;
  } else if (num > MAX[2]) MAX[2] = num;
}

int main() {
  ifstream fin("input.txt");
  string s;
  int curr = 0;
  while (getline(fin, s)) {
    if (s.empty()) {
      mx(curr);
      curr = 0;
      continue;
    } else curr += stoi(s);
  }
  mx(curr);
  cout << (MAX[0] + MAX[1] + MAX[2]) << '\n';
  return 0;
}

File: solutions/advent_of_code/2022/Day 2/part2.py
# Advent of Code Problem Link: https://adventofcode.com/2022/day/2

with open('input.txt', 'r') as file:
  inpt = file.readlines()
  file.close()
inpt = [i.strip() for i in inpt]
map_input = {
  'A': 'Rock',
  'B': 'Paper',
  'C': 'Scissors',
  'X': 'Lose',
  'Y': 'Draw',
  'Z': 'Win',
}
shape_points = {
  'Rock': 1,
  'Paper': 2,
  'Scissors': 3,
}
round_points = {
  'Lose': 0,
  'Draw': 3,
  'Win': 6,
}
total_score = 0
for i in inpt:
  opponent = map_input[i[0]]
  goal = map_input[i[2]]
  total_score += round_points[goal]
  if (opponent, goal) in [('Rock', 'Draw'), ('Paper', 'Lose'), ('Scissors', 'Win')]:
    total_score += shape_points['Rock']
  elif (opponent, goal) in [('Paper', 'Draw'), ('Scissors', 'Lose'), ('Rock', 'Win')]:
    total_score += shape_points['Paper']
  else:
    total_score += shape_points['Scissors']
print(total_score)
